ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/BaseTransactionManagerDelegate.java

```diff
@@ -336,6 +336,11 @@ public void unlock(final TransactionLocal local, final Transaction transaction)
     // Bad things will probably happen if users ever use this key themselves
     private final String LOCKS_MAP = "__LOCKS_MAP";
 
+    // Using string LOCKS_MAP directly as object of synchronization is not recommended
+    // String literals are centrally interned and could also be locked on by a library,
+    // potentially having deadlocks or lock collisions
+    private final Object locksMapLock = new Object();
```

Comment:
+1 
You need locksMapLock to be a static final Object, because of the reason the static analysis tool gave ("_LOCKS_MAP" as a final string is interned so same object used by other threads).